### PR TITLE
Micoprofile config

### PR DESCRIPTION
**Pull request by mikecroft**

### CODE REVIEW

File: documentation/microprofile/README.adoc [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+|
+* JAX-RS 2.0
+* CDI 1.1
+* JSON-P 1.0
+* http://microprofile.io/project/eclipse/microprofile-config[Config 1.0]
+
+| [source, xml]
+----
+<dependency>
+    <groupId>org.eclipse.microprofile</groupId>
```
Comment (OndrejM) [inline documentation/microprofile/README.adoc]: MP 1.1 hasn't been released and this is still subject to change. Personally, I would change artifactId and type if possible and everybody agrees.

File: documentation/microprofile/config.adoc [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+== Background
+The Config API was created to allow for the modification of application
+configuration from outside the application without needing the application to be
+repackaged.
+
+A full overview for the reasoning behind the API can be found in
+https://github.com/eclipse/microprofile-config/blob/master/README.adoc[the
+README for the source code repository.]
+
+https://github.com/eclipse/microprofile-config/blob/master/spec/src/main/asciidoc/microprofile-config-spec.asciidoc[
```
Comment (OndrejM) [inline documentation/microprofile/config.adoc]: I would rather link to the Config 1.0 release page: https://github.com/eclipse/microprofile-config/releases/tag/1.0
It contains Javadoc, PDF spec and other info.

File: documentation/microprofile/config.adoc [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+|Source
+|Ordinal
+|Description
+
+|`domain`
+|110
+|Retrieves and stores config values in the domain.xml as key values. These will
+then apply to all instances within a domain
+
+|`config`
```
Comment (OndrejM) [inline documentation/microprofile/config.adoc]: To be honest, from this description, I have no idea how this works

Comment (OndrejM) [inline documentation/microprofile/config.adoc]: Part of it is that the sentence doesn't make sense. Should it be 2 separate sentences?

File: documentation/microprofile/config.adoc [hunk 4]
```diff
@@ -0,0 +1,302 @@
...
+configuration
+
+|`server`
+|130
+|Retrieves and stores config values at the level of the server in the domain.xml.
+Property values will only apply on the named server instance
+
+|`application`
+|140
+|retrieves and stores config values at the level of the application. Property
```
Comment (OndrejM) [inline documentation/microprofile/config.adoc]: Captial R in **R**etrieves and stores

File: documentation/microprofile/config.adoc [hunk 5]
```diff
@@ -0,0 +1,302 @@
...
+
+== Converters
+In addition to the standard converters specified by the API, Payara's implementation
+includes out-of-the-box converters for URL and InetAddress. These have `@Priority`
+of `1`, the same as the default converters.
+
+
+== Asadmin Commands
+
+IMPORTANT: A `--source` value of `jndi` is not supported for this command in
```
Comment (OndrejM) [inline documentation/microprofile/config.adoc]: I would move this notice under the *Usage* section of the set-config-ordinal` command. In this place it is rather confusing to which command it relates to.

File: documentation/microprofile/README.adoc (hunk 1 above)
Comment (OndrejM) [inline documentation/microprofile/README.adoc]: I mean, we shouldn't change this now, but update it once MP 1.1 artifact is finally released.

File: documentation/microprofile/config.adoc (hunk 5 above)
Comment (mikecroft) [inline documentation/microprofile/config.adoc]: It relates to all commands. It was originally there, but when I noticed it was missing from everywhere, I moved it. Should be reworded.

Comment (OndrejM) [inline documentation/microprofile/config.adoc]: As it is now, it looks pretty scary - before reading any useful documentation, the reader hits an IMPORTANT warning...

I would rather move it at the end of the page, if it applies to all commands and also reword it to make sense. It;s not a big deal because it should be fixed in the next version, but the psychological impact could be big if it stays like this.

File: documentation/microprofile/README.adoc (hunk 1 above)
Comment (mikecroft) [inline documentation/microprofile/README.adoc]: Agreed

File: documentation/microprofile/README.adoc [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+* JAX-RS 2.0
+* CDI 1.1
+* JSON-P 1.0
+
+| [source, xml]
+----
+<dependency>
+    <groupId>io.microprofile</groupId>
+    <artifactId>microprofile</artifactId>
+    <version>1.0.0</version>
```
Comment (mikecroft) [inline documentation/microprofile/README.adoc]: changed

File: documentation/microprofile/config.adoc (hunk 3 above)
Comment (mikecroft) [inline documentation/microprofile/config.adoc]: Clarified
